File: php/graphql/field/calc.php
<?php
/**
 * User: WindomZ
 * Date: 17-7-20
 */

namespace GraphQLExamples\Field;

use GraphQL\Type\Definition\Type;

function getSumField()
{
    return [
        'type' => Type::int(),
        'args' => [
            'x' => ['type' => Type::int()],
            'y' => ['type' => Type::int()],
        ],
        'resolve' => function ($root, $args) {
            return $args['x'] + $args['y'];
        },
    ];
}

File: go/graphql/field/calc.go
package graphqlexamples

import (
	"errors"

	"github.com/graphql-go/graphql"
)

// SumField define a sum mutation field
var SumField *graphql.Field

func init() {
	SumField = &graphql.Field{
		Type: graphql.Int,
		Args: graphql.FieldConfigArgument{
			"x": &graphql.ArgumentConfig{
				Type: graphql.Int,
			},
			"y": &graphql.ArgumentConfig{
				Type: graphql.Int,
			},
		},
		Resolve: func(p graphql.ResolveParams) (interface{}, error) {
			x, ok := p.Args["x"].(int)
			if !ok {
				return "Can not read 'x'!", nil
			}
			y, ok := p.Args["y"].(int)
			if !ok {
				return nil, errors.New("Can not read 'y'!")
			}
			return x + y, nil
		},
	}
}
